### ISSUE TITLE
building uTox on Win7

### ISSUE THREAD

**Issue by pugongying** [uTox/uTox#1172]
### Overview

1. uTox version:
2. Operating system: **win7 32bit**
3. Desktop Environment/Window Manager (Linux/BSD only):

### Issue

When building uTox on Win7, I build it step by step,by Windows section of uTox/docs/BUILD.md,but after typing the command "cmake -DCMAKE_TOOLCHAIN_FILE="../cmake/toolchain-win32.cmake" -DTOXCORE_STATIC=ON .."meet a problem.
    I hope for your help.Thank your very much!

**-- The C compiler identification is GNU 6.4.0
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc.exe
-- Check for working C compiler: /usr/bin/i686-w64-mingw32-gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- checking for C compiler flag: -Wall
-- checking for C compiler flag: -Wextra
-- checking for C compiler flag: -Wpointer-arith
-- checking for C compiler flag: -Wimplicit-fallthrough=5
-- Warning: Can't find flag -Wimplicit-fallthrough=5
-- checking for C compiler flag: -Werror=implicit-function-declaration
-- checking for C compiler flag: -Wformat=0
-- checking for C compiler flag: -Wno-misleading-indentation
-- checking for C compiler flag: -fno-strict-aliasing
-- checking for C compiler flag: -fPIC
-- checking for C compiler flag: -static
-- checking for C compiler flag: -DAUDIO_FILTERING=1
-- Found PkgConfig: /usr/bin/pkg-config.exe (found version "0.29.1")
CMake Error at /usr/share/cmake-3.6.2/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find libtox (missing: LIBTOX_LIBRARIES)
Call Stack (most recent call first):
  /usr/share/cmake-3.6.2/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  cmake/Findlibtox.cmake:53 (find_package_handle_standard_args)
  CMakeLists.txt:275 (find_package)**

**Comment by zoff99** [comment]
i have the same issue

**Comment by redmanmale** [comment]
Probably it's because build instructions aren't yet updated to use ToxCore 0.2.x.
I'll take look.

**Comment by zoff99** [comment]
@redmanmale can you add a circleCI windows build?

**Comment by redmanmale** [comment]
@zoff99 
Why? CircleCI doesn't support Windows and we already have Travis for cross-compiling.
I'm working on setting up the AppVeyor build (it supports Windows) but could not get precise ETA for this.

**Comment by zoff99** [comment]
@redmanmale because circleCI has the ability to save artefacts and make them downloadable via browser

**Comment by redmanmale** [comment]
> artefacts downloadable via browser

It's configurable.

CircleCI doesn't build on Windows. What's the diffrence from Travis?

**Comment by robinlinden** [comment]
To build uTox against Toxcore downloaded from build.tox.chat, you need https://github.com/uTox/uTox/pull/1157.

**Comment by zoff99** [comment]
@redmanmale if its configurable in travis. please configure it so that the resulting binary is downloadable.
would be cool

**Comment by pugongying** [comment]
Thank you all ! I'm sorry for never used CircleCI and Travis, and my native language is Chinese. I'll learn about CircleCI and Travis.

**Comment by redmanmale** [comment]
CircleCI, AppVeyor and Travis configurations are out of scope for this.

This will be fixed after merge of #1157.

**Comment by redmanmale** [comment]
This problem cannot be solved until PR (#1157) will be merged.
Because it brings support of new version of the ToxCore.

For now to build uTox you have to get ToxCore v.0.1.11 (build it yourself).

**Comment by fcore117** [comment]
What is blocking currently to merge new? any errors in code still remaining until uTox can move forward with core development?

I personally tested 0.2x core patched uTox and it compiles and runs fine (Win 10) do not know about Win 7 but should be ok. Maybe someone can confirm that if it is ok on Win 7

**Comment by robinlinden** [comment]
Now that #1157 is merged, this should work. :)

**Comment by ArkYk** [comment]
Hi,@robinlinden
I have the same problem after #1157 ,could you please help me?

cmake -DCMAKE_TOOLCHAIN_FILE="../cmake/toolchain-win64.cmake" -DTOXCORE_STATIC=ON ..
-- The C compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc.exe
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- checking for C compiler flag: -Wall
-- checking for C compiler flag: -Wextra
-- checking for C compiler flag: -Wpointer-arith
-- checking for C compiler flag: -Wimplicit-fallthrough=5
-- checking for C compiler flag: -Werror=implicit-function-declaration
-- checking for C compiler flag: -Wformat=0
-- checking for C compiler flag: -Wno-misleading-indentation
-- checking for C compiler flag: -fno-strict-aliasing
-- checking for C compiler flag: -fPIC
-- checking for C compiler flag: -static
-- checking for C compiler flag: -DAUDIO_FILTERING=1
-- Found PkgConfig: /usr/bin/pkg-config.exe (found version "0.29.1")
CMake Error at /usr/share/cmake-3.6.2/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find libtox (missing: LIBTOX_LIBRARY LIBTOX_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.6.2/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  cmake/Findlibtox.cmake:26 (find_package_handle_standard_args)
  CMakeLists.txt:274 (find_package)


-- Configuring incomplete, errors occurred!
See also "/cygdrive/c/projects/uTox/build/CMakeFiles/CMakeOutput.log".

**Comment by redmanmale** [comment]
> Could NOT find libtox (missing: LIBTOX_LIBRARY LIBTOX_INCLUDE_DIR)

Ensure that you have all required binary files of Toxcore v.2.x (and others) in `/cygdrive/c/projects/libs/windows-x64`

**Comment by redmanmale** [comment]
Maybe there's a bug in build guide for Windows; will check it later.

**Comment by redmanmale** [comment]
There's a mistake in build guide which will be fixed in #1301.

**Comment by ArkYk** [comment]
@redmanmale Thank you for your help！ :D 
I'd like to make a few Suggestions：
    1.the uTox version always changing,in the Build.md,I want the files link are the latest what your last build passing.
    2.about "Could NOT find libtox (missing: LIBTOX_LIBRARY LIBTOX_INCLUDE_DIR)",I find it need install in cgywin,then it‘s passing.remain "Could NOT find libtox (missing: LIBTOX_LIBRARY LIBTOX_INCLUDE_DIR)"

**Comment by ArkYk** [comment]
> 构建指南中存在错误，将在[＃1301中](https://github.com/uTox/uTox/pull/1301)修复。

@redmanmale maybe you can see that＃1305 .

**Comment by ArkYk** [comment]
https://github.com/uTox/uTox/issues/1305#issue-373331719
